GitCommands/Git/GitModule.cs

```diff
@@ -611,22 +611,30 @@ public bool HandleConflictsSaveSide(string fileName, string saveAsFileName, stri
 
         public void SaveBlobAs(string saveAs, string blob)
         {
-            using (var blobStream = GetFileStream(blob))
+            try
             {
-                var blobData = blobStream.ToArray();
-                if (EffectiveConfigFile.core.autocrlf.Value == AutoCRLFType.@true)
+                using (var blobStream = GetFileStream(blob))
                 {
-                    if (!FileHelper.IsBinaryFileName(this, saveAs) && !FileHelper.IsBinaryFileAccordingToContent(blobData))
+                    var blobData = blobStream.ToArray();
+                    if (EffectiveConfigFile.core.autocrlf.Value == AutoCRLFType.@true)
                     {
-                        blobData = GitConvert.ConvertCrLfToWorktree(blobData);
+                        if (!FileHelper.IsBinaryFileName(this, saveAs) && !FileHelper.IsBinaryFileAccordingToContent(blobData))
+                        {
+                            blobData = GitConvert.ConvertCrLfToWorktree(blobData);
+                        }
                     }
-                }
 
-                using (var stream = File.Create(saveAs))
-                {
-                    stream.Write(blobData, 0, blobData.Length);
+                    using (var stream = File.Create(saveAs))
+                    {
+                        stream.Write(blobData, 0, blobData.Length);
+                    }
                 }
             }
+            catch (IOException ex)
+            {
+                // currently only handling IOException in client code
+                throw ex;
```

Comment:
❗  This will alter the call stack!

This will retain the original stack:
```suggestion
                throw;
```

Reply:
Done!